test/framework/src/main/java/org/elasticsearch/test/rest/ESRestTestCase.java

```diff
@@ -668,6 +684,11 @@ private void wipeCluster() throws Exception {
 
         wipeSnapshots();
 
+        if (preserveSystemFeatureStates() == false) {
```

Comment:
Could we use positive logic? I think it makes these flags easier to think about. That is, `resetSystemFeatureStates()` which defaults to `true`, and methods override it to disable resetting.

Reply:
Sounds good.